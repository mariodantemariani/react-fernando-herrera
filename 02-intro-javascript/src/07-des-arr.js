const personajes = ["Goku", "Vegeta", "Trunks"];

const [, , p3] = personajes;
console.log(p3);

const retornaArreglo = () => {
  return ["ABC", 123];
};

const [letras, numeros] = retornaArreglo();
console.log(letras, numeros);

// Tarea
// 1. el primer valor del arr se llamará nombre
// 2. se llamará setNombre
const seState = (valor) => {
  return [
    valor,
    () => {
      console.log("Hola Mundo");
    },
  ];
};

const [nombre, setNombre] = seState("Goku");

console.log(nombre);
setNombre();

const usarState = (valor) => {
  return [
    valor,
    () => {
      console.log("Hola Mundo");
    },
  ];
};

const [nombre2, setNombre2] = usarState("Tony");
console.log(nombre2, setNombre2);
