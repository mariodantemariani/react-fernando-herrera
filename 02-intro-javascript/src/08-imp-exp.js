import heroes from "./data/heroes";

export const getHeroeById = (id) => {
  return heroes.find((heroe) => heroe.id === id);
};

export const getHeroesByOnwner = (owner) => {
  return heroes.filter((heroe) => heroe.owner === owner);
};

//console.log(getHeroeById(2));
//console.log(getHeroesByOnwner("DC"));
