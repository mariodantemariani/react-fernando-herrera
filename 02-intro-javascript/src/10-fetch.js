const apiKey = "fTt8q1qIDlo6mEPmnDQvS5a9K6vsdCwm";
const endPoint = "http://api.giphy.com/v1/gifs/random?api_key=";

const peticion = fetch(`${endPoint}${apiKey}`);

peticion
  .then((res) => res.json())
  .then(({ data }) => {
    const { url } = data.images.original;
    const img = document.createElement("img");
    img.src = url;

    document.body.append(img);
  })
  .catch(console.warn);
