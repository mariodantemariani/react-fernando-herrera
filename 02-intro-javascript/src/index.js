const apiKey = "fTt8q1qIDlo6mEPmnDQvS5a9K6vsdCwm";
const endPoint = "http://api.giphy.com/v1/gifs/random?api_key=";

// const getImagenPromesa = () => new Promise( resolve => resolve('https://ajskdhaskjdhajs.com') )
// getImagenPromesa().then( console.log );

const getImagen = async () => {
  try {
    const resp = await fetch(`${endPoint}${apiKey}`);
    const { data } = await resp.json();

    const { url } = data.images.original;

    const img = document.createElement("img");
    img.src = url;
    document.body.append(img);
  } catch (error) {
    console.error(error);
  }
};

getImagen();
