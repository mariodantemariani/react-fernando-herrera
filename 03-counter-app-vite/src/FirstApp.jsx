import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle }) => {
  return (
    <>
      <h1>{title}</h1>
      <h2>{subTitle + 1}</h2>
      <p>Soy un subtitulo</p>
    </>
  );
};

FirstApp.propTypes = {
  subTitle: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  subTitle: 11,
  title: "no hay",
};
