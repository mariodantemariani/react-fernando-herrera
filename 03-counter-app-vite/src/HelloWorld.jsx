export function App() {
  return <h1>Hola Mundo!!</h1>;
}

export const HelloWorld = () => {
  return <div>HelloWorld</div>;
};
