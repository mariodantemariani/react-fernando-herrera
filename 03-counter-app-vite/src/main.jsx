import React from "react";
import ReactDOM from "react-dom/client";
//import { App, HelloWorld } from "./HelloWorld";
//import { FirstApp } from "./FirstApp";
import { CounterApp } from "./CounterApp/CounterApp";

import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <App />
    <HelloWorld /> */}
    {/* <FirstApp title="Hola, soy Gku" /> */}
    <CounterApp value={0} />
  </React.StrictMode>
);
