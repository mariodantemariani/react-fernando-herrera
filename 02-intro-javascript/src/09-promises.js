import { getHeroeById, getHeroesByOnwner } from "./08-imp-exp";

// const promesa = new Promise((resolve, reject) => {
//   setTimeout(() => {
//     const heroe = getHeroeById(2);
//     resolve(heroe);
//     //reject("no se pudo manejar el error");
//   }, 2000);
// });

// promesa
//   .then((variable) => {
//     console.log("then promesa", variable);
//   })
//   .catch((err) => console.log(err));

const getHeroeByIdAsync = (id) => {
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      const heroe = getHeroeById(id);
      if (heroe) {
        resolve(heroe);
      } else {
        reject("no se encontro");
      }
    }, 2000);
  });
};

getHeroeByIdAsync(114)
  .then((heroe) => console.log("Heroe", heroe))
  .catch((err) => console.warn(err));

// promesa
//   .then((variable) => {
//     console.log("then promesa", variable);
//   })
//   .catch((err) => console.log(err));
