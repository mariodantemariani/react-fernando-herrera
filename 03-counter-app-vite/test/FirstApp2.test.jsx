import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en First App", () => {
  const title = "Test title";
  const subTitle = "No hay subtÃ­tulo";

  test("should be the same that snapShot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("should shown the title in the h1-using contain", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("should shown the title in the h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("should shown the subtitle sent by props", () => {
    render(<FirstApp title={title} subTitle={subTitle} />);

    expect(screen.getAllByText(subTitle).length).toBe(1);
  });
});
