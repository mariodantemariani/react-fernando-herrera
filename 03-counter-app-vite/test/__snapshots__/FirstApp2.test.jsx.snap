// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en First App should be the same that snapShot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Test title
     
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
